### ISSUE TITLE
Executive Members

### ISSUE THREAD

**Issue by asmahood** [uwpokerclub/website#383]
## Summary

The website has no concept of an executive member. Executives and ordinary members are the same row in the same table, which forces workarounds and produces user-visible bugs.

Executives get free memberships, so nobody sets their `paid` flag. That leaves them at `paid = false`, which is indistinguishable from a member who simply hasn't paid — and `paid = false` is the trigger for free-trial logic. Executives therefore accumulate toward `semesters.free_trial_limit`, get shaded red in the members list and the event registration modal, and — the real breakage — are **blocked from signing into events** with `"Membership has no free trial events remaining"` (`server/internal/services/participants_service.go:92`).

The root cause is that one boolean, `paid`, is being asked to mean two different things: *"this person gave the club money"* and *"this person is not subject to the trial limit."* Executives are the case where those two come apart.

The workaround is already load-bearing in the UI: `MembershipConfig.tsx` carries the comment *"This section is only for executive members"* — a note about who operates the form, sitting directly above the checkboxes that misclassify them.

## Goals

1. Executives are a first-class, explicitly-set property of a membership.
2. Executives are never subject to free-trial logic — not the sign-in block, not the red shading, not the cached flag.
3. Executives are visually distinguishable and filterable in the members list.
4. Paying / discounting an executive membership is not offered, and is rejected if attempted.

## Non-goals

- **Executive job titles** (President, Treasurer, …) on the membership. `logins.role` already models the admin-permission hierarchy; duplicating titles onto memberships would create a second source of truth with no consumer yet.
- **Any coupling between `logins.role` and the membership flag.** Considered and explicitly dropped — see Design decisions.
- **Changing authorization.** Marking someone an executive member grants no website permissions. Permissions remain entirely a function of `logins.role`.
- **Replacing `paid`/`discounted` with a status enum.** The right long-term model, but a breaking migration across the API contract, rankings queries, filters, and E2E fixtures.
- **Backfilling existing data.** See Rollout.
- **Rankings changes.** Executives appear in rankings exactly as they do today.

## Design decisions

| # | Decision | Choice | Rationale |
|---|---|---|---|
| 1 | Where `executive` lives | A boolean column on **`memberships`** (per-semester), not on `users` and not derived. | There is a new set of executives each term, and memberships are already the per-semester entity. A flag on `users` would retroactively rewrite past semesters when someone leaves the exec team. |
| 2 | Coupling to `logins.role` | **None.** Purely a manual checkbox. | Executives are added to a semester by a President/VP/Secretary, or by themselves, and those people already know who the executives are. A login-derived hint would have added a `LEFT JOIN logins ON logins.username = users.quest_id` and a new computed API field to save one click, while leaning further on that implicit convention. |
| 3 | Relationship to `paid`/`discounted` | **Exclusive.** `executive = true` implies `paid = false` and `discounted = false`. Executive is its own membership status. | Matches the real world — executives are comped, not paying — and keeps a single readable status per membership rather than a matrix. |
| 4 | Budget impact | **$0.** Creating an executive membership skips `IncrementBudget` entirely. | Follows from #3. Executives pay nothing, so nothing enters the budget. |
| 5 | Toggling `executive` on an already-paid membership | Auto-clear `paid`/`discounted` and **reverse the budget credit** for the fee that was recorded. | Symmetric with the reversal `UpdateMembership` already performs when un-paying (`membership_service.go:102`), so no new budget arithmetic is introduced. Un-marking `executive` leaves the membership Unpaid with no credit. |
| 6 | Members-list treatment | **Badge + filter.** An "Executive" value in the Status column, plus an Executive option in the filter drawer. No grouping, tabs, or second table. | Smallest change that satisfies "displayed separately," and it reuses the filter drawer that already exists. |
| 7 | Filter shape | **Additive** — a new `executive` param alongside the existing `paid` and `discounted`, rather than consolidating all three into one `status` param. | Keeps existing `?paid=&discounted=` URLs and Cypress specs working. Cost: the drawer can express combinations that cannot exist, mitigated by disabling Paid/Discounted while Executive is active. |
| 8 | Implementation shape | Additive column plus **one shared eligibility predicate** replacing three duplicated `!Paid` checks. | The free-trial rule is currently written out three times; adding `&& !Executive` to each would preserve exactly the duplication that produced this bug. |
| 9 | Backfill | **None.** Column defaults to `false`; current executives are corrected by hand. | Roughly eight rows, once, and every future term starts clean. |

## Data model

```sql
ALTER TABLE memberships
  ADD COLUMN executive boolean NOT NULL DEFAULT false;
```

**Invariant** (enforced in `membershipService`, not by a DB constraint — consistent with how `paid`/`discounted` consistency is enforced today):

> `executive == true` ⇒ `paid == false && discounted == false`

**Derived status:**

| `executive` | `paid` | `discounted` | Status |
|---|---|---|---|
| `true` | `false` | `false` | **Executive** |
| `false` | `true` | `true` | Paid (Discounted) |
| `false` | `true` | `false` | Paid |
| `false` | `false` | `false` | Unpaid |

**The shared predicate:**

```go
// EligibleForFreeTrial reports whether the free-trial limit applies to this
// membership at all. Paid members have bought unlimited entry; executives are
// comped and were never on a trial to begin with.
func (m Membership) EligibleForFreeTrial() bool {
    return !m.Paid && !m.Executive
}
```

**Note on `free_trial_available` for executives:** `syncFreeTrialAvailable` returns early for executives, so an executive's stored `free_trial_available` keeps whatever value it last held. This is safe because every read is gated behind `EligibleForFreeTrial()`, but it means a raw SQL query on that column alone is misleading for executives. The column was already documented as a display-only cache that can go stale; this widens that caveat rather than introducing a new class of problem.

## Work breakdown

Child issues are tracked as native sub-issues; ordering is tracked via GitHub's issue dependency graph (see each issue's "Blocked by" section).

**Backend:**
- [ ] #384 — Add `executive` to memberships and exempt executives from free-trial logic. **Closes the sign-in breakage on its own**, before any UI exists.
- [ ] #385 — Create/update semantics and budget handling (blocked by #384)
- [ ] #386 — Filter memberships by executive status (blocked by #384)

**Frontend:**
- [ ] #387 — Executive checkbox in the membership form (blocked by #385)
- [ ] #388 — Executive badge, list filter, and trial-shading exemption (blocked by #386)

```
#384  model + free-trial predicate
├─ #385  create/update + budget
│   └─ #387  membership form checkbox
└─ #386  executive filter
    └─ #388  badge + filter + shading
```

## Rollout

1. Ship the migration (additive, `DEFAULT false` — no downtime, no data rewrite).
2. Post-deploy, an executive opens Members for the current semester and ticks Executive on the current exec team (roughly eight rows). Their red trial shading clears immediately, and any sign-in block lifts.
3. No backfill script, no data migration, no feature flag. Future terms start clean because executives are registered before the term's first event.

## Risks

| Risk | Assessment |
|---|---|
| Someone forgets to tick the box for a new exec | Same failure mode as today, but now with a visible fix. The exec appears as Unpaid and eventually gets trial-shaded — a legible symptom pointing at the right control. |
| Stale `free_trial_available` on executives | Safe because every read is gated behind the predicate. Worth a comment at the column and at `syncFreeTrialAvailable`. |
| Executives in rankings | Explicitly unchanged. If the club later decides executives should be excluded from prize rankings, that is a separate issue — this design neither enables nor blocks it, and the flag is the natural hook. |
| The exclusive-status model | If the club ever wants an executive who *also* paid, decision 3 makes that unrepresentable. Judged not to occur; decision 5's refund path is the escape hatch. |
